addons/mail/static/src/core/common/attachment_list.js

```diff
@@ -112,9 +112,8 @@ export class AttachmentList extends Component {
     get showDelete() {
         return (
             (this.attachment.isDeletable &&
-                (!this.env.message || this.env.message?.hasTextContent)) ||
-            this.env.inComposer ||
-            this.props.attachments.length > 1
```

Comment:
There are mostly 3 ways attachment list is used:
- in composer
- on message
- in chatter attachment box / channel attachment panel

The `!this.env.message` was specifically there to handle the latter case. By removing it, it becomes impossible to delete attachment when there's a single one in the chatter attachment box:

![Screenshot 2024-05-14 at 11 12 14 (2)](https://github.com/odoo/odoo/assets/6569390/98ce4063-5476-4cc3-a09e-8e48f733f40a)

So this `!this.env.message` condition should be kept

Reply:
Ah, I think I understand the use case now. So maybe it didn't check "deletable" because it was implicit in this context then I guess.

Then would something like this work?
- If composer, always show delete
- If attachments are not deletable, never show
- If attachments are linked to a message, the message is empty and there's only one attachment. Do not show

Reply:
> Then would something like this work?
> 
> * If composer, always show delete
> * Otherwise If attachments are not deletable, never show
> * If attachments are linked to a message, the message is empty and there's only one attachment. Do not show

Yes, this is correct!